packages/flutter_tools/lib/src/build_system/targets/android.dart

```diff
@@ -229,7 +230,10 @@ class AndroidAot extends AotElfBase {
     if (!output.existsSync()) {
       output.createSync(recursive: true);
     }
-    final List<String> extraGenSnapshotOptions = decodeDartDefines(environment.defines, kExtraGenSnapshotOptions);
+    final List<String> extraGenSnapshotOptions = List.from(decodeDartDefines(environment.defines, kExtraGenSnapshotOptions));
+    if (environment.defines[kSplitAot] == 'true') {
+      extraGenSnapshotOptions.add('--loading_unit_manifest=${output.path}/manifest.json');
```

Comment:
use platform file separator?

Reply:
This also needs to be in the output files. Is there anything wrong with always generating it?

Reply:
This option is the switch that determines if gen_snapshot produces split aot or not. WIthout, it produces a regular aot snapshot.